homeassistant/components/overkiz/entity.py

```diff
@@ -119,3 +119,5 @@ def __init__(
             # In case of sub device, use the provided label
             # and append the name of the type of entity
             self._attr_name = f"{self.device.label} {description.name}"
+        else:
+            self._attr_name = description.name
```

Comment:
Casting something to string meant that both `None` and `UndefinedType` (if no name was set) got turned into a string. I removed that statement and now type checking is correctly complaining.

It is better to just set `name=None` in the entity descriptions that need it.

Reply:
I will need to have a better look here... Setting `name` to `None` in entity description won't work, because we actually want to use the name from the entity description.

There are other cases where we want to inherit the name from the device name, for example for cover entities. Since a recent change in core, it will print an error when we implicit inherit the name from the device. Looking at this at a first glance, it will require more changes to explicit set the name.

Reply:
I have made a fix for now. However I should have a better look in the coming days (or shortly after the next release), to see if we can rewrite the `OverkizEntity`.